src/actions/arcActions.ts

```diff
@@ -186,11 +191,15 @@ export async function getDAOData(avatarAddress: string, currentAccountAddress: s
   const executedProposals = await (await contributionRewardInstance.getExecutedProposals(daoInstance.avatar.address))({}, { fromBlock, toBlock }).get();
   const proposals = [...votableProposals, ...executedProposals];
 
-  // Get all proposals' details like title and description from the server
+  // Get all proposals' details like title and url from the server
   let serverProposals: { [key: string]: any } = {};
   try {
-    const results = await axios.get(process.env.API_URL + '/api/proposals?filter={"where":{"daoAvatarAddress":"' + avatarAddress + '"}}');
-    serverProposals = _.keyBy(results.data, "arcId");
+    const promises = proposals.map(async (proposal) => {
+      const ipfsHash = ipfs.hexToHash(proposal.contributionDescriptionHash)
+      const result = await ipfs.get(ipfsHash)
+      return result
+    })
+    serverProposals = await Promise.all(promises)
```

Comment:
I think we should change the use of the word "server" everywhere to something else, maybe metaProposals? better ideas?

Reply:
"proposals"

Reply:
Proposals are built from both off-chain details and on-chain parameters so we'd probably want to reserve the `proposals` name for the complete chunk of proposal data.
"proposalDetails" is good but conflicts w existing vars so might be a pain to incorporate.
"offChainProposalDetails" seems most descriptive although a bit wordy.

Reply:
+1 for wordy vars :-)